network: com.docker.network.bridge.inhibit_ipv4 (#19833)
docker/docs: content/network/drivers/bridge.md

| ------------------------------------------------ | -------------- | -------------------------------------------------------------------------------------------- |
| `com.docker.network.bridge.name`                 |                | Interface name to use when creating the Linux bridge.                                        |
| `com.docker.network.bridge.enable_ip_masquerade` | `true`         | Enable IP masquerading.                                                                      |
| `com.docker.network.bridge.enable_icc`           | `true`         | Enable or Disable inter-container connectivity.                                              |
| `com.docker.network.bridge.host_binding_ipv4`    |                | Default IP when binding container ports.                                                     |
| `com.docker.network.driver.mtu`                  | `0` (no limit) | Set the containers network Maximum Transmission Unit (MTU).                                  |
| `com.docker.network.container_iface_prefix`      | `eth`          | Set a custom prefix for container interfaces.                                                |

> | `com.docker.network.bridge.inhibit_ipv4`         | `false`        | Prevent Docker from [assigning a gateway IP](#skip-ip-gateway-configuration) to the network. |

Reviewer: Ah, previous version was mentioning 'unmanaged bridge'. I think it'd be great to better document that feature and make the section about 'inhibit_ipv4' part of it.

Author: yeah, I guess that's what I had in mind when I started writing, but then I realized all I'm documenting now is the IP address configuration, so I had to constrain the scope of my section slightly. Maybe later!

> | `com.docker.network.bridge.inhibit_ipv4`         | `false`        | Prevent Docker from [assigning a gateway IP](#skip-ip-gateway-configuration) to the network. |

Reviewer: ```suggestion
| `com.docker.network.bridge.inhibit_ipv4`         | `false`        | Prevent Docker from [assigning an IP address](#skip-ip-address-configuration) to the bridge. |
```

For more information about this limitation, see
[moby/moby#44973](https://github.com/moby/moby/issues/44973#issuecomment-1543747718).

> ## Skip IP gateway configuration

Reviewer: ```suggestion
## Skip IP address configuration
```

The `com.docker.network.bridge.inhibit_ipv4` option lets you create a network

> that uses an existing bridge and have Docker skip configuring the gateway IP on
> the bridge. This is useful if you want to configure the IP gateway for the
> bridge manually.

Reviewer: ```suggestion
that uses an existing bridge and have Docker skip configuring the IPv4 address on
the bridge. This is useful if you want to configure the IP address for the
bridge manually, for instance if you add a physical interface to your bridge and need to move its IP address to the bridge interface.
```

For more information about this limitation, see
[moby/moby#44973](https://github.com/moby/moby/issues/44973#issuecomment-1543747718).

## Skip IP address configuration

The `com.docker.network.bridge.inhibit_ipv4` option lets you create a network
that uses an existing bridge and have Docker skip configuring the IPv4 address
on the bridge. This is useful if you want to configure the IP address for the
bridge manually. For instance if you add a physical interface to your bridge,
and need to move its IP address to the bridge interface.

To use this option, you should first configure the Docker daemon to use a
self-managed bridge, using the `bridge` option in the `daemon.json` or the
`dockerd --bridge` flag.

With this configuration, north-south traffic won't work unless you've manually

> configured the IP gateway for the bridge.

Reviewer: ```suggestion
configured the IP address for the bridge.
```
